### DISCUSSION TITLE
Specify global data in frontmatter

### DISCUSSION THREAD

**Discussion by freshyill** [11ty/eleventy#2074] [Q&A] [answered]
I'm trying to specify the name of a global data file in frontmatter. It looks like this:

```yaml
---
kingdom: insects
---
```
```njk

{% for animal in kingdom %}
  {{ animal.name }}
{% endfor %}
```

Here `insects` is a reference to `_data/insects.json`. The problem is that "insects" is interpreted as a string (which makes sense). 

How can I insert the value of `kingdom` as if I'm calling the name of global data file?

One thing to clarify: I'm **not** trying to [create pages from data](https://www.11ty.dev/docs/pages-from-data/). I'm just trying to display data *on* pages. Also, this is Nunjucks, if that matters. Thanks!

**Comment by pdehaan** [comment, marked as answer]
@freshyill I can think of two possible solutions:

1. You could keep your structure as-is, and probably use `eleventyComputed` to dynamically resolve `kingdom` from the /_data/ directory, like so:
	  ```js
	  ---js
	  {
	    key: "insects",
	    eleventyComputed: {
	      kingdom(data) {
	        return data[data.key];
	      }
	    }
	  }
	  ---
	  
	  {% for animal in kingdom %}
	    <p>{{ animal.name }}</p>
	  {% endfor %}
	  ```
	  
	  Possibly better ways of doing that. I sometimes find JavaScript frontmatter confusing, but it works.

2. Slight tweak to your /_data/ directory, but if you move the insects.json file into a subfolder, like /_data/dynamic/insects.json, you can reference it dynamically like so:
	  ```njk
	  ---
	  kingdom: insects
	  ---
	  
	  {% for animal in dynamic[kingdom] %}
	    <p>{{ animal.name }}</p>
	  {% endfor %}
	  ```

**Reply by freshyill** [reply]
Thanks, moving the files was a great, simple solution!
